### PR TITLE
Ana/add the ledger windows stuff from old lunie

### LINKED ISSUE #248: Ledger Nano S not working on Win10/Chrome
I believe I saw a workaround for this on lunie.io.. Would it be possible to make it work in lunie light as well?

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #248

### CODE REVIEW

File: pages/ledger/index.vue [hunk 1]
```diff
@@ -9,6 +9,33 @@
...
+      Due to recent Ledger updates, using a Ledger on Windows now requires
+      "Experimental Web Platform features" to be enabled.
+      <div v-if="isChrome">
+        <br />
+        <br />
+        <p>
+          Please copy the link below into a new tab and set the "Experimental
+          Web Platform features" flag to "Enabled":
+        </p>
+        <div
```
Comment (faboweb) [inline pages/ledger/index.vue:22]: please refactor into a copyable link... this is just duplicated code

File: pages/ledger/index.vue [hunk 2]
```diff
@@ -9,6 +9,33 @@
         @click.native="connect"
       />
     </div>
+    <div v-if="isWindows && !hasHIDEnabled">
+      Due to recent Ledger updates, using a Ledger on Windows now requires
+      "Experimental Web Platform features" to be enabled.
```
Comment (faboweb) [inline pages/ledger/index.vue:14]: this is chrome dependant so isChrome should be included in the v-if
there is also a v-else missing explaining that Ledger will not work and the user should use Chrome or Brave

Comment (Bitcoinera) [reply pages/ledger/index.vue:14]: oh I see. Thank you

File: pages/ledger/index.vue (hunk 1 above)
Comment (Bitcoinera) [reply pages/ledger/index.vue:22]: you mean CopyButton? I don't see any other reusable component for this

File: store/ledger.js [hunk 3]
```diff
@@ -25,7 +27,7 @@ export const actions = {
   async init({ commit }) {
     commit('setLoading', true)
     try {
-      const ledger = await getLedger()
+      const ledger = await getLedger(state().isWindows, state().hasHIDEnabled)
```
Comment (faboweb) [inline store/ledger.js:30]: get state from first argument

File: common/ledger.js [hunk 4]
```diff
@@ -1,18 +1,34 @@
 import { getHDPath } from './hdpath'
 import network from './network'
 
-export async function getLedger() {
+export async function getLedger(isWindows, hasHIDEnabled) {
```
Comment (faboweb) [inline common/ledger.js:4]: why pass this in here? you can calculate both variables in here right?

**Comment by Bitcoinera** [comment]
Now I am getting this all the time...

image

File: common/ledger.js (hunk 4 above)
Comment (Bitcoinera) [reply common/ledger.js:4]: well, that is true

**Comment by faboweb** [comment]
I guess you can't open WebHID twice. Store the object and close the connection if it already exists?

**Comment by Bitcoinera** [comment]
> I guess you can't open WebHID twice. Store the object and close the connection if it already exists?

I tried avoiding to create the transport the second time but didn't work. I don't know exactly what is happening 🤷

**Comment by Bitcoinera** [comment]
It is still quite strange... I need to refresh every time I do some action with the Ledger (login & sending transaction...). Otherwise, the next action I get a `Device is already open` error

**Comment by Bitcoinera** [comment]
I don't know :shrug:  Someone has some idea? Only works refreshing on Windows... otherwise it throws the `Device already open` error after each action :shrug:

**Comment by Bitcoinera** [comment]
Good to go! Everything working

**Comment by jbibla** [comment]
can you share screenshots of the new warnings?

**Comment by Bitcoinera** [comment]
> can you share screenshots of the new warnings?

The new warnings? Well they are the same than in old Lunie, so not really new. And I already shut down my Windows and hope I don't need to use it ever again xD

**Comment by Bitcoinera** [comment]
> can you share screenshots of the new warnings?

OK, but "cheating", this is the Windows warning:

image


Thisis the Linux / OSX one:

image

File: common/ledger.js [hunk 5]
```diff
@@ -33,12 +33,12 @@ export async function getLedger() {
...
     )
   }
-  const ledger = new LedgerSigner(transport, {
+  const ledger = new LedgerSigner(ledgerTransport || transport, {
     testModeAllowed: true,
     hdPaths: [await getHDPath(network.HDPath)],
     prefix: network.addressPrefix,
   })
-  return ledger
+  return { ledger, transport }
```
Comment (faboweb) [inline common/ledger.js:44]: tip:

```
return {
   getAccounts() {
     const accounts = await ledger.getAccounts()
     transport.disconnect()
     return accounts
   },
   ....
}
```

Comment (faboweb) [reply common/ledger.js:44]: but for a refactor

**Review by faboweb — APPROVED** [review]

**Pull request merged**